cmd/entrypoint.go

```diff
@@ -42,11 +44,12 @@ func EntrypointCommand(cmd *cobra.Command, args []string) error {
 		execwrap.ExecWrap().RunSyscallExec(args)
 	} else {
 		err := execwrap.ExecWrap().RunCommandDirect(args)
-		if err != nil {
-			return err
-		}
-
-		err = CleanCommand(cmd, filesToDecrypt)
+		defer func(parentErr error) {
```

Comment:
Any reason to put this in the `err` check directly instead of in `defer`?

Reply:
I honestly debated it as it seemed a bit of a stretch case for defer.

Regarding doing the cleanup inside the `err` check: I didn't want to duplicate the code both inside and outside the check in case there _wasn't_ an error as cleanup must be done in both cases.

I originally had the defer right after the call to `DecryptCommand`. This was before I changed the return to a named return and at that point I wasn't able to capture the actual error. Since I added the named return (`rerr`) I could adjust all the other return points to set that which would allow the defer to be placed immediately after the decryption where it belongs. This would allow me to push the original error all the way out. I ultimately didn't do it this way because in the `exec3` case the deferred cleanup is not triggered and it seemed silly to cover both cases.

I went ahead and prepped that in case you prefer it: https://github.com/Ibotta/sopstool/compare/master...elementalvoid:bug/cleanup-on-entrypoint-error-2

Or ... something else entirely : ) I'm open to your thoughts on direction.

Reply:
oh right that totally makes sense! I didn't realize we weren't autocleaning after entrypoint already.

Putting the defer outside of the if branch (just after decrypt) does make  sense though - exec3 will return an error if the command isn't executable, etc. so it would be good to clean up even in that case. This seems like a good way to handle it.

Thanks for the good thinking!

Reply:
Cool. I'll update first thing tomorrow.

Reply:
Having the defer modify the function value returned is not a pattern I have seen in the docs.  It took a few extra cycles to understand the behavior. I recommend adding a comment or a test.
It did force me to understand when values passed to a defer are evaluated. :)

Reply:
@onyxraven 
> exec3 will return an error if the command isn't executable

Sounds like I should go ahead and capture this error for return too. Currently we just .. don't.